Source/santactl/Commands/sync/SNTCommandSyncManager.m

```diff
@@ -103,6 +107,34 @@ - (instancetype)initWithDaemonConnection:(SNTXPCConnection *)daemonConn isDaemon
 
 #pragma mark SNTSyncdXPC protocol methods
 
+- (void)postBundleEventToSyncServer:(SNTStoredEvent *)event reply:(void (^)(BOOL))reply {
+  SNTCommandSyncState *syncState = [self createSyncState];
+  SNTCommandSyncEventUpload *p = [[SNTCommandSyncEventUpload alloc] initWithState:syncState];
+  if (event && [p uploadEvents:@[event]]) {
+    BOOL needsRelatedEvents = [syncState.bundleBinaryRequests containsObject:event.fileBundleHash];
+    reply(needsRelatedEvents);
+    if (needsRelatedEvents) {
+      LOGD(@"Needs related events");
+    } else {
+      LOGD(@"Bundle event upload complete");
+    }
+  } else {
+    reply(NO);
+    LOGE(@"Bundle event upload failed");
+  }
+}
+
+- (void)postBundleEventsToSyncServer:(NSArray<SNTStoredEvent *> *)events {
+  SNTCommandSyncState *syncState = [self createSyncState];
+  syncState.eventBatchSize = 50;
```

Comment:
Will do so in the next pull request. We can hold it in santactl since it runs as a daemon now.